code/modules/power/singularity/emitter.dm

```diff
@@ -361,6 +370,64 @@
 	user?.visible_message("[user.name] emags [src].",span_notice("You short out the lock."))
 
 
+//Emiter with much higher structural damage but less normal one
+/obj/machinery/power/emitter/drill
+	name = "drill"
+	desc = "A heavy-duty industrial laser, modified to destroy structures and rocks."
+	laser_damage = 5
+	var/laser_structural_damage = 100
+	projectile_type = /obj/projectile/beam/emitter/drill/hitscan
+
+/obj/machinery/power/emitter/drill/RefreshParts()
+	var/max_fire_delay = 12 SECONDS
+	var/fire_shoot_delay = 12 SECONDS
+	var/min_fire_delay = 2.4 SECONDS
+	var/las_damage = 0
+	var/las_structural_damage = 50
+	var/power_usage = 350
+	for(var/obj/item/stock_parts/micro_laser/laser in component_parts)
+		max_fire_delay -= 2 SECONDS * laser.rating
+		min_fire_delay -= 0.4 SECONDS * laser.rating
+		fire_shoot_delay -= 2 SECONDS * laser.rating
+		las_damage += 5 * laser.rating
+		las_structural_damage += 50 * laser.rating
+	laser_damage = las_damage
+	maximum_fire_delay = max_fire_delay
+	minimum_fire_delay = min_fire_delay
+	fire_delay = fire_shoot_delay
+	laser_structural_damage = las_structural_damage
+	for(var/obj/item/stock_parts/manipulator/manipulator in component_parts)
+		power_usage -= 50 * manipulator.rating
+	update_mode_power_usage(ACTIVE_POWER_USE, power_usage)
+
+/obj/machinery/power/emitter/drill/fire_beam(mob/user)
```

Comment:
I haven't compared yet, but this likely has a lot of redundancy with `/obj/machinery/power/emitter/fire_beam()` as well. 

Anything you copy pasted exactly from the original proc does not need to be re-entered in a child, it will already run the same proc with the same information. You only need to define **new** things.